cluster/buckets.go

```diff
@@ -175,17 +176,54 @@ func ListBuckets(tenantShortname string) ([]TenantBucketInfo, error) {
 	return bucketInfos, nil
 }
 
-// Deletes a bucket in the given tenant's MinIO
-func DeleteBucket(tenantShortname, bucket string) error {
+// DeleteBucket Deletes a bucket in the given tenant's MinIO
+func DeleteBucket(ctx *Context, bucket string) error {
 	// Get tenant specific MinIO client
-	minioClient, err := newTenantMinioClient(nil, tenantShortname)
+	minioClient, err := newTenantMinioClient(ctx, ctx.Tenant.ShortName)
 	if err != nil {
 		return err
 	}
-
+	// Verify if bucket is being used within a permission.
+	// 	If bucket is being used, we don't allow the deletion.
+	//  The permission needs to be updated first.
+	bucketUsed, err := bucketInPermission(ctx, bucket)
+	if err != nil {
+		return err
+	}
+	if bucketUsed {
+		return fmt.Errorf("bucket is being used in at least one permission")
+	}
 	return minioClient.RemoveBucket(bucket)
 }
 
+// bucketInPermission returns wether a bucket is being used within a permission or not
+func bucketInPermission(ctx *Context, bucketName string) (bool, error) {
+	query := `
+		SELECT 
+			COUNT(*)
+		FROM 
+			permissions_resources
+		WHERE 
+		    bucket_name = $1`
```

Comment:
you could speed up this by introducing `LIMIT 1`

Reply:
and adding an index to the `bucket_name` field

Reply:
makes sense. Done